src/test/java/com/google/finapp/FinAppIT.java

```diff
@@ -124,4 +129,154 @@ public void createAccount_createsSingleValidAccount() throws Exception {
       assertThat(count).isEqualTo(1);
     }
   }
+
+  @Test
+  public void createTransactionForAccount_isCredit_subtractsFromAccountBalance() throws Exception {
+    ByteArray accountId = UuidConverter.getBytesFromUuid(UUID.randomUUID());
+    BigDecimal oldAccountBalance = new BigDecimal(20);
+    BigDecimal amount = new BigDecimal(10);
+    boolean isCredit = true;
+    databaseClient.write(
+        ImmutableList.of(
+            Mutation.newInsertBuilder("Account")
+                .set("AccountId")
+                .to(accountId)
+                .set("AccountType")
+                .to(AccountType.UNSPECIFIED_ACCOUNT_TYPE.getNumber())
+                .set("AccountStatus")
+                .to(AccountStatus.UNSPECIFIED_ACCOUNT_STATUS.getNumber())
+                .set("Balance")
+                .to(oldAccountBalance)
+                .set("CreationTimestamp")
+                .to(Value.COMMIT_TIMESTAMP)
+                .build()));
+    BigDecimal outputNewBalance =
+        spannerDao.createTransactionForAccount(accountId, amount, isCredit);
+    assertThat(outputNewBalance).isEqualTo(oldAccountBalance.subtract(amount));
+    try (ReadOnlyTransaction transaction = databaseClient.readOnlyTransaction();
+        ResultSet transactionResultSet =
+            transaction.read(
+                "TransactionHistory",
+                KeySet.all(),
```

Comment:
We get `all` here instead of a specific key because the "TransactionHistory" table is keyed under timestamp and account id (have to include both for a `key`), and we don't know the exact time of the commit. I couldn't figure out a method with these keys/keysets for setting the account id but allowing the timestamp to be anything, so I just grab everything.